File: mvvm/src/main/java/com/hhh/mvvm/base/BaseFragment.java
package com.hhh.mvvm.base;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.IdRes;
import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.hhh.mvvm.listener.OnActivityResultListener;
import com.hhh.mvvm.listener.OnPageSelectedListener;
import com.trello.rxlifecycle3.components.support.RxFragment;

public abstract class BaseFragment extends RxFragment implements OnPageSelectedListener {

  public BaseFragment() {
  }

  @LayoutRes
  protected abstract int getLayoutResId();

  @Nullable
  @Override
  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
    @Nullable Bundle savedInstanceState) {
    int layoutId = getLayoutResId();
    if (layoutId != 0) {
      return inflater.inflate(layoutId, container, false);
    }
    return null;
  }

  @Nullable
  public final <T extends View> T findViewById(@IdRes int id) {
    return getView() != null ? getView().findViewById(id) : null;
  }

  public final void finish() {
    if (isAdded()) {
      requireActivity().finish();
    }
  }

  public final void onBackPressed() {
    if (isAdded()) {
      requireActivity().onBackPressed();
    }
  }

  public final void startActivityForCallback(@NonNull Intent intent, int requestCode,
    @NonNull OnActivityResultListener resultListener) {
    startActivityForCallback(intent, requestCode, null, resultListener);
  }

  public final void startActivityForCallback(@NonNull Intent intent, int requestCode,
    @Nullable Bundle options, @NonNull OnActivityResultListener resultListener) {
    if (isAdded()) {
      BaseActivity activity = (BaseActivity) requireActivity();
      activity.startActivityForCallback(intent, requestCode, options, resultListener);
    }
  }
}

File: core/src/main/java/com/nb/core/BaseFragment.kt
package com.nb.core

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope

abstract class BaseFragment : Fragment(), OnPageSelectedListener {

    val viewLifecycle by lazy { viewLifecycleOwner.lifecycle }
    val viewLifecycleScope by lazy { viewLifecycleOwner.lifecycleScope }

    protected abstract val layoutId: Int

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? = if (layoutId != 0) inflater.inflate(layoutId, container, false) else null

    fun finishActivity() {
        if (isAdded) {
            requireActivity().finish()
        }
    }

    fun onBackPressed() {
        if (isAdded) {
            requireActivity().onBackPressed()
        }
    }
}